### DISCUSSION TITLE
How to customize the template

### DISCUSSION THREAD

**Discussion by sharoonthomas** [retypeapp/retype#372] [Q&A]
How do you go about customizing the theme?

Here is what I want to do:

1. Add a UUID to the meta on the .md pages
2. Use the UUID to create a permalink and allow copying the URL from the page

Any recommendations on how to go about something like this?

**Comment by geoffreymcgill** [comment]
Unfortunately, there is no ability to customize the theme, although you can use a custom `data` config to pass the value into the page.

```md
---
data:
    uuid: "123456"
---
# Sample page

[{{ uuid }}](sample.md?{{ uuid }})
```

> Use the UUID to create a permalink and allow copying the URL from the page

Could you demonstrate the Markdown or HTML that you would like to see added to the page(s)? and where on the page

**Reply by sharoonthomas** [reply]
Thanks Geoffrey

Something like a button, which on clicking copies a permanent link.

Like what github does on source code:
![image](https://user-images.githubusercontent.com/305807/183509970-5759a606-0d73-4578-9742-50088f483a76.png)

**Reply by geoffreymcgill** [reply]
Each heading on the page does have something similar, where hovering on the heading will show the anchor hash that can be right-clicked on.

<img width="262" alt="Screen Shot 2022-08-08 at 2 45 04 PM" src="https://user-images.githubusercontent.com/62210/183511287-b0a663fc-6c68-411b-ba3a-665dada92d0c.png">

Building something similar to your **Copy permalink** screen capture would not be possible, at least with the built-in functionality of Retype.

**Reply by sharoonthomas** [reply]
Yeah that was just an example.

![image](https://user-images.githubusercontent.com/305807/183511865-6f2e6c86-7d57-4522-bf1a-db90b45b3353.png)

Would it be possible to add it here? or perhaps we should write this completely in JS?
